### DISCUSSION TITLE
Noob question: I'm building a LSP integration into Monaco -- what does this library do/ how can it help me do this?

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [CodinGame/monaco-vscode-api#415] [Q&A]
I'm currently using this repository as an example: https://github.com/erictraut/pyright-playground/tree/main

Which sets up a simple Express server with the LSP and then the client uses `registerCompletionItemProvider` to make API calls as needed to get suggestions.

Would monaco-vscode-api make this an easier process / would it make it more resilient?

**Comment by CGNonofr** [comment]
It seems you are reimplementing the language server protocol by hands on the client. This library allows to use the `vscode-languageclient` library directly on the client, you just need to implement the channel with the server (websockets is a privileged way, using npmjs.com/package/vscode-ws-jsonrpc)

**Reply by kaisalmen** [reply]
@jordyvandomselaar you can also use [monaco-languageclient](https://github.com/TypeFox/monaco-languageclient). This project was created by @CGNonofr to overcome its limitations: https://github.com/CodinGame/monaco-vscode-api#history

There is a fully working python client and server example available:
https://github.com/TypeFox/monaco-languageclient#main-examples

**Reply by jordyvandomselaar** [reply]
Thank you for the replies both! I think we'll need to stick with our current manual implementation since we're horizontally scaling the LSP for many users but this is helpful! =)
